plugin/pkg/up/up.go

```diff
@@ -53,12 +54,21 @@ func (p *Probe) start(interval time.Duration) {
 			// Passed the lock. Now run f for as long it returns false. If a true is returned
 			// we return from the goroutine and we can accept another Func to run.
 			go func() {
+				i, j := 1, 1
 				for {
 					if err := f(); err == nil {
 						break
 					}
-					// TODO(miek): little bit of exponential backoff here?
-					time.Sleep(interval)
+					time.Sleep(time.Duration(i) * interval)
+					// Every third failure, we increase the interval, but only up to 4 * interval.
+					j++
+					if j%3 == 0 {
+						i++
+						if i > 4 {
+							i = 4
+						}
+					}
+
```

Comment:
If I look only to the diff, then the implementation looks ok.

But, I am wondering what happen when the f() always return an error (because the remote endpoint is down), and you Stop or Reload the plugin. => this "go func()" will run forever and continue to run f() on the same remote endpoint no ?


=> you may end-up with weird behavior with obsolete code still called in background. Especially in the the RELOAD use case. 

Am I right ?

Reply:
Should we add a mechanism to get out of this"go func()" when the Proxy is deleted ?

Reply:
This is stopped when this is called: https://github.com/coredns/coredns/blob/master/plugin/forward/proxy.go#L82